src/lib/pages/backoffice/DocumentRequest/DocumentRequestDetails/DocumentRequestSteps/DocumentRequestStepPanels/ChooseDocumentStepPanel/ChooseDocumentStepPanel.js

```diff
@@ -15,36 +15,33 @@ export default class ChooseDocumentStepPanel extends Component {
   };
 
   goToCreateDocumentAndPrefill = (docReq) => {
-    const data = {};
-    const authors = _get(docReq, 'authors');
-    if (authors) {
-      data['authors'] = [{ full_name: _get(docReq, 'authors') }];
-    }
-    const journalTitle = _get(docReq, 'journal_title', '');
-    const issue = _get(docReq, 'issue', '');
-    const volume = _get(docReq, 'volume', '');
-    if (journalTitle || issue || volume) {
-      data['publication_info'] = [
-        {
-          journal_title: journalTitle,
-          journal_issue: issue,
-          journal_volume: volume,
-        },
-      ];
-    }
-    const isbn = _get(docReq, 'isbn');
-    if (isbn) {
-      data['identifiers'] = [{ scheme: 'ISBN', value: isbn }];
-    }
+    const {
+      authors,
+      journal_title: journalTitle,
+      issue,
+      volume,
+      isbn,
+      edition,
+      publication_year,
+      publisher,
+    } = docReq;
 
-    const edition = _get(docReq, 'edition');
-    if (edition) {
-      data['edition'] = edition;
-    }
-    const publicationYear = _get(docReq, 'publication_year');
-    if (publicationYear) {
-      data['publication_year'] = `${publicationYear}`; // it has to be a string
-    }
+    const data = {
+      authors: authors && [{ full_name: authors }],
+      publication_info: _some([journalTitle, issue, volume])
+        ? [
+            {
+              journal_title: journalTitle ?? '',
+              journal_issue: issue ?? '',
+              journal_volume: volume ?? '',
+            },
+          ]
+        : [],
+      identifiers: isbn ? [{ scheme: 'ISBN', value: isbn }] : [],
+      edition: edition,
+      publication_year: publication_year?.toString(),
+      imprint: { publisher: publisher ?? '' },
```

Comment:
I would say let's match the code style and behavior that was before in place just to have some consistency on the code i.e in the way we check for values existence.